src/utils/handleSubmit.js

```diff
@@ -0,0 +1,43 @@
+import Popup from "../components/Popup.js";
+
+export function handleSubmit(
+  submitAction,
+  popupInstance, // This should be an instance of Popup or PopupDeleteCard
+  submitText = "Save",
+  loadingText = "Saving..."
+) {
+  // Check if popupInstance is an instance of Popup
+  if (!(popupInstance instanceof Popup)) {
+    console.error("The provided popupInstance is not an instance of Popup.");
+    return Promise.reject("Invalid popup instance");
+  }
+
+  const popup = popupInstance._popupElement;
+  const submitButton = popup.querySelector(".modal__save-button");
+
+  if (!submitButton) {
+    console.error("Submit button not found in popup.");
+    return Promise.reject("Submit button not found");
+  }
+
+  submitButton.disabled = true;
+  submitButton.textContent = loadingText;
+
+  // Ensure submitAction returns a promise
+  return Promise.resolve(submitAction())
+    .then(() => {
+      popupInstance.close();
+      submitButton.disabled = false;
+      submitButton.textContent = submitText;
```

Comment:
`  submitButton.textContent = submitText;` is not needed here because you have `this.renderLoading(true);` inside `PopupWithForm`

Or you can call `renderLoading` here no to duplicate the logics. It’s a violation of **Single responsibility** principle